components/inputs/Autocomplete.tsx

```diff
@@ -39,39 +47,43 @@ function AutocompleteInner<O>(
 
   const handleChange = useCallback(
     async (event: NativeSyntheticEvent<TextInputChangeEventData>) => {
-      console.log({ event })
       const newInput = event.nativeEvent.text
       setInput(newInput)
       debouncedHandleChange(newInput)?.catch((error) => false)
     },
     [debouncedHandleChange],
   )
 
-  const handleOptionPress = (option: O) => () => {
-    setValue(option)
-    setInput(getOptionLabel(option))
-    onValueChange(option)
-    setOptions([])
-  }
+  const handleOptionPress = useCallback(
+    (option: O) => () => {
+      setValue(option)
+      setInput(getOptionLabel(option))
+      onValueChange(option)
+      setOptions([])
+    },
+    [getOptionLabel, onValueChange],
+  )
+
+  const defaultRenderItem: ListRenderItem<O> = useCallback(
+    ({ item }) => (
+      <PressableStyled
+        className="border-divider p-3"
+        // eslint-disable-next-line react-native/no-inline-styles
+        style={{ borderBottomWidth: 1 }}
+        onPress={handleOptionPress(item)}
+      >
+        <Typography>{getOptionLabel(item)}</Typography>
+      </PressableStyled>
+    ),
+    [handleOptionPress, getOptionLabel],
+  )
 
   return (
     <View ref={ref}>
       {/* eslint-disable-next-line @typescript-eslint/no-misused-promises */}
       <TextInput onChange={handleChange} value={input} />
       <View>
-        <FlatList
-          renderItem={({ item }) => (
-            <PressableStyled
-              className="border-divider p-3"
-              // eslint-disable-next-line react-native/no-inline-styles
-              style={{ borderBottomWidth: 1 }}
-              onPress={handleOptionPress(item)}
-            >
-              <Typography>{getOptionLabel(item)}</Typography>
-            </PressableStyled>
-          )}
-          data={options}
-        />
+        <FlatList renderItem={renderItem || defaultRenderItem} data={options} />
```

Comment:
`||` will take all "falsy" values, such as empty string or `0`
In this case, I would use `??` operator, to check only if `renderItem` is `undefined` and `null`.

Generally, `??` is usually what we want to use. And use `||` just in specific cases where you need this specific behaviour.